simulide: add generic builder for other versions

TomaSajt opened the pull request:
## Description of changes

@CarlosCraveiro

This is a continuation of https://github.com/NixOS/nixpkgs/pull/273641, which only added the `0_4_15` version of this package.

This PR separates simulide into 3 versions: `simulide_0_4_15`, `simulide_1_0_0` and `simulide_1_1_0`
Just writing `simulide` will use `simulide_1_0_0` as that is latest version marked as stable.

The git diff is quite horrible, the indentation messes it up a lot.



<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

CarlosCraveiro commented:
Okay, I tested the new package and everything seems working as it should! By the way, sorry about the messed indentation. I yet have not fully configured my editor, so some things ends messed up (I think it was the case of the indentation at the files from the old PR).

CarlosCraveiro reviewed (approved):
LGTM!

TomaSajt commented:
>By the way, sorry about the messed indentation. I yet have not fully configured my editor, so some things ends messed up (I think it was the case of the indentation at the files from the old PR).

Actually, that PR was fine. The issue was that this `generic` builder had to be indented like 2 extra levels compared to the other PR. This made git confused what lines came from where, and now it thinks it got completely rewritten. But it's nothing to worry about.

NickCao commented on pkgs/top-level/all-packages.nix:
```diff
@@ -39673,7 +39673,10 @@ with pkgs;
   appcsxcad = libsForQt5.callPackage ../applications/science/electronics/appcsxcad { };
 
-  simulide = libsForQt5.callPackage ../applications/science/electronics/simulide { };
+  inherit (libsForQt5.callPackage ../applications/science/electronics/simulide { })
```
Ideally we should use callPackages (notices the s) here for override to properly work, but I don't know how to make it work with libsForQt5.

TomaSajt replied:
Maybe I could separate the versions into different small `.nix` files, and call them one-by-one by path in `all-packages.nix`. Though I am not sure if that would solve the issue completely, because the generic builder currently uses packages from the same place for all versions.

NickCao reviewed (commented):
Please split the changes into individual commits.

1. add the generic builder and convert the existing version
2. one commit for each new version

TomaSajt committed:
simulide: add generic builder, set simulide alias to simulide_0_4_15

TomaSajt committed:
simulide_1_0_0: init at 1.0.0-SR2, set simulide alias to simulide_1_0_0

TomaSajt committed:
simulide_1_1_0: init at 1.1.0-RC1

TomaSajt commented:
I am not very proud of the commit names.
Should I have added a `simulide_0_4_15: init at 0.4.15-SR10` part somewhere? It's not a new package, just renamed, but with an alias...
